File: VisualBasicAssignment/screen/sai_keat/widget/order/CartScreen.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace VisualBasicAssignment
{
    public partial class CartScreen
    {
        public CartScreen()
        {
            InitializeComponent();
            _btnPay.Name = "btnPay";
        }

        private List<CartDetail> details;

        private void CartScreen_Load(object sender, EventArgs e)
        {
            pnlPicture.AutoScroll = true;
            ReLoad();
        }

        private void ReLoad()
        {
            pnlPicture.Controls.Clear();
            using (var db = new FoodShopEntities1())
            {
                int id = DatabaseConnections.GetLoginUser().user_id;
                var cart_record = db.Carts.Where(s => s.customer_id == id).FirstOrDefault();
                if (cart_record is null)
                {
                    details = ExFunctions.CreateCart().CartDetails.ToList();
                }
                else
                {
                    details = cart_record.CartDetails.ToList();
                }

                try
                {
                    for (int i = 0, loopTo = details.Count; i <= loopTo; i++)
                    {
                        var items = new OrderItem(details[i]);
                        items.Visible = true;
                        Console.WriteLine("Order Id: {0}", details[i].food_id);
                        pnlPicture.Controls.Add(items);
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void btnPay_Click(object sender, EventArgs e)
        {
            PaymentDetail pd;
            using (var db = new FoodShopEntities1())
            {
                OrderItem ctrl;
                var total = default(double);
                var newPaynment = new Payment();
                newPaynment.PaymentStatu = db.PaymentStatus.Find((object)Guid.Parse("20bd9f32-b531-4b7f-a89a-80babe9aa707"));
                newPaynment.user_id = DatabaseConnections.GetLoginUser().user_id;
                newPaynment.payment_date = DateTime.Now;
                newPaynment.id = Guid.NewGuid();
                newPaynment.notes = "";
                newPaynment.payment_method = Guid.Parse("9244ee14-b4b5-4f2c-833b-3b3c18d68764");
                newPaynment.payment_meta = "";
                for (int i = 0, loopTo = details.Count - 1; i <= loopTo; i++)
                {
                    ctrl = (OrderItem)pnlPicture.Controls[i];
                    if (ctrl.ckb.Checked)
                    {
                        total += Convert.ToDouble(details[i].Food.price * details[i].quantity);
                        pd = new PaymentDetail();
                        pd.add_date = DateTime.Now;
                        pd.quantity = details[i].quantity;
                        pd.food_id = details[i].food_id;
                        pd.payment_id = newPaynment.id;
                        db.CartDetails.Remove(db.CartDetails.Find(details[i].cart_id, details[i].food_id));
                        newPaynment.PaymentDetails.Add(pd);
                    }
                }

                newPaynment.net_pay = total;
                newPaynment.tax_change = total * 0.06d;
                newPaynment.total_pay = total * 1.06d;
                db.Payments.Add(newPaynment);
                try
                {
                    db.SaveChanges();
                    DatabaseConnections.SetCurrentPayment(newPaynment);
                    My.MyProject.Forms.PaymentStep1.Show();
                    ReLoad();
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }
    }
}

File: VisualBasicAssignment/screen/sai_keat/widget/order/CartScreen.vb
Public Class CartScreen
    Dim details As List(Of CartDetail)
    Private Sub CartScreen_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        Me.pnlPicture.AutoScroll = True
        ReLoad()
    End Sub

    Private Sub ReLoad()
        pnlPicture.Controls.Clear()
        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Dim id As Int32 = DatabaseConnections.GetLoginUser().user_id
            Dim cart_record As Cart = db.Carts.Where(Function(s) s.customer_id = id).FirstOrDefault()

            If cart_record Is Nothing Then
                details = ExFunctions.CreateCart().CartDetails.ToList()
            Else
                details = cart_record.CartDetails.ToList()
            End If
            Try
                For i = 0 To details.Count
                    Dim items As OrderItem = New OrderItem(details.Item(i))
                    items.Visible = True
                    Console.WriteLine("Order Id: {0}", details.Item(i).food_id)
                    pnlPicture.Controls.Add(items)
                Next
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using
    End Sub

    Private Sub btnPay_Click(sender As Object, e As EventArgs) Handles btnPay.Click
        Dim pd As PaymentDetail

        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Dim ctrl As OrderItem
            Dim total As Double
            Dim newPaynment As Payment = New Payment()
            newPaynment.PaymentStatu = db.PaymentStatus.Find(Guid.Parse("20bd9f32-b531-4b7f-a89a-80babe9aa707"))
            newPaynment.user_id = DatabaseConnections.GetLoginUser().user_id
            newPaynment.payment_date = DateTime.Now
            newPaynment.id = Guid.NewGuid()
            newPaynment.notes = ""
            newPaynment.payment_method = Guid.Parse("9244ee14-b4b5-4f2c-833b-3b3c18d68764")
            newPaynment.payment_meta = ""

            For i = 0 To details.Count() - 1
                ctrl = CType(pnlPicture.Controls.Item(i), OrderItem)
                If (ctrl.ckb.Checked) Then
                    total += Convert.ToDouble(details.Item(i).Food.price * details.Item(i).quantity)
                    pd = New PaymentDetail()
                    pd.add_date = Date.Now
                    pd.quantity = details.Item(i).quantity
                    pd.food_id = details.Item(i).food_id
                    pd.payment_id = newPaynment.id
                    db.CartDetails.Remove(db.CartDetails.Find(details.Item(i).cart_id, details.Item(i).food_id))
                    newPaynment.PaymentDetails.Add(pd)
                End If
            Next

            newPaynment.net_pay = total
            newPaynment.tax_change = total * 0.06
            newPaynment.total_pay = total * 1.06
            db.Payments.Add(newPaynment)

            Try
                db.SaveChanges()
                DatabaseConnections.SetCurrentPayment(newPaynment)
                PaymentStep1.Show()
                ReLoad()
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using
    End Sub
End Class
